src/index.js

```diff
@@ -262,30 +262,18 @@ class SignatureFactory {
     this.method = method;
   }
 
-  getCanonicalHeaders () {
-    return {
-      date: `date:${this.dates.usbldate}`,
-      host: `host:${this.host}\n`
-    };
-  }
-
   hash (string, encoding) {
     return crypto.createHash('sha256').update(string, 'utf8').digest(encoding);
   }
 
   canonicalString () {
-    const queryStr = this.queryParameters;
-    const bodyHash = this.hash('', 'hex');
-    const canonicalHeaders = this.getCanonicalHeaders();
-
     return [
-      this.method || 'GET',
-      this.url,
-      queryStr,
-      canonicalHeaders.date,
-      canonicalHeaders.host,
-      'date;host',
-      bodyHash
+      this.method || 'GET',               // HTTPRequestMethod
+      this.url,                           // CanonicalURI
+      this.queryParameters,               // CanonicalQueryString
+      `host:${this.host}`+ '\n' + `x-usbl-date:${this.dates.longdate}`+ '\n', // CanonicalHeaders
+      'host;x-usbl-date',                 // SignedHeaders
```

Comment:
hmm, this is difficult..
You suggestion does not hold, as the headers need to be in a specific place / order based on the name. Therefore if extracting the date header name,  one will loose the contextual connection to the order.
I thought about grouping all headers (and order) in one place, but this can be approached from different aspects, as some headers are signature related and others are not. Also the content of the header (the longdate) is defined in two places, which makes me sad.

I suggest to leave it as it is, and some extensive discussion with refactoring needs to be done to solve this.